File: frontend/angular-drinkShop/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {

    imageUrl: string;
    price: number;
    quantity: number;
    drinkId: number;

    constructor(cartItem: CartItem) {
        this.imageUrl = cartItem.imageUrl;
        this.price = cartItem.price;
        this.quantity = cartItem.quantity;
        this.drinkId = cartItem.id;
    }
}

File: backend/spring-boot-drinkShop/src/main/java/com/fullStackProject/drinkShop/entity/OrderItem.java
package com.fullStackProject.drinkShop.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "order_item")
@Getter
@Setter
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "price")
    private double price;

    @Column(name = "drink_id")
    private int drinkId;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;
}
